pycbc/detector.py

```diff
@@ -94,36 +100,61 @@ def add_detector_on_earth(name, longitude, latitude,
         # assume right angle detector if no separate xarm direction given
         xangle = yangle + np.pi / 2.0
 
+    # Rotation matrix to move detector to correct orientation
+    rm1 = rotation_matrix(longitude * units.rad, 'z')
+    rm2 = rotation_matrix((np.pi / 2.0 - latitude) * units.rad, 'y')
+    rm = np.matmul(rm2, rm1)
+
     # Calculate response in earth centered coordinates
     # by rotation of response in coordinates aligned
     # with the detector arms
-    a, b = cos(2*xangle), sin(2*xangle)
-    xresp = np.array([[-a, b, 0], [b, a, 0], [0, 0, 0]])
-    a, b = cos(2*yangle), sin(2*yangle)
-    yresp = np.array([[-a, b, 0], [b, a, 0], [0, 0, 0]])
-    resp = (yresp - xresp) / 4.0
+    resps = []
+    vecs = []
+    for angle in [yangle, xangle]:
+        a, b = cos(2 * angle), sin(2 * angle)
+        resp = np.array([[-a, b, 0], [b, a, 0], [0, 0, 0]])
 
-    rm1 = rotation_matrix(longitude * units.rad, 'z')
-    rm2 = rotation_matrix((np.pi / 2.0 - latitude) * units.rad, 'y')
-    rm = np.matmul(rm2, rm1)
+        # apply rotation
+        resp = np.matmul(resp, rm)
```

Comment:
Optional point, but it might be nicer to use the matrix multiplication operator `@`.